rustls-mbedpki-provider/src/tests_common.rs

```diff
@@ -12,8 +12,20 @@ use core::{
     ops::{Deref, DerefMut},
 };
 
-use rustls::pki_types::{CertificateDer, PrivateKeyDer, ServerName, UnixTime};
 use rustls::{client::danger::ServerCertVerifier, ClientConnection, ConnectionCommon, ServerConnection, SideData};
+use rustls::{
+    crypto::{ring, CryptoProvider},
+    pki_types::{CertificateDer, PrivateKeyDer, ServerName, UnixTime},
+};
+
+use std::sync::Once;
+
+pub(crate) fn init_crypto_provider() {
```

Comment:
If we don't do this, would the `aws-lc-rs` provider be used by default?

Reply:
No, becuase some feature issue, the test will throw error that no default provider is available.
I choose to use `ring`, because `aws-lc-rs` have a c library dependency (something like `aws-lc-rs-sys` similar to our `mbedtls-sys-auto`) which is hard to maintain.

Reply:
Wait, I guess I met some wired thing about feature, Now the code can automatically choose to use `ring` provider successfully without this function. I am going to remove it.

Ref:
https://docs.rs/rustls/latest/src/rustls/crypto/mod.rs.html#254-284